tests/integration/targets/hcloud_image_info/tasks/test.yml

```diff
@@ -1,92 +1,104 @@
-    id: "{{hcloud_test_image_id}}1"
-    type: snapshot
-  ignore_errors: true
+    name: "{{ hcloud_image_name }}"
   register: result
-- name: verify test gather hcloud image with wrong id
-  assert:
+- name: Verify hcloud_image_info with correct name
+  ansible.builtin.assert:
     that:
-      - result is failed
+      - result.hcloud_image_info | list | count == 1
+      - result.hcloud_image_info[0].architecture == "x86"
 
-- name: test gather hcloud image infos with name
+- name: Gather hcloud_image_info with wrong name
   hetzner.hcloud.hcloud_image_info:
-    name: "{{ hcloud_test_image_name_os }}"
-  register: hcloud_images
-- name: verify test gather hcloud image infos with name
-  assert:
+    name: "{{ hcloud_image_name }}-invalid"
+  register: result
+- name: Verify hcloud_image_info with wrong name
+  ansible.builtin.assert:
     that:
-      - hcloud_images.hcloud_image_info | list | count == 1
-      - hcloud_images.hcloud_image_info[0].architecture == "x86"
+      - result.hcloud_image_info | list | count == 0
 
-- name: test gather hcloud image infos with name and architecture
+- name: Gather hcloud_image_info with correct name and architecture
   hetzner.hcloud.hcloud_image_info:
-    name: "{{ hcloud_test_image_name_os }}"
+    name: "{{ hcloud_image_name }}"
     architecture: arm
-  register: hcloud_images
-- name: verify test gather hcloud image infos with name
-  assert:
+  register: result
+- name: Verify hcloud_image_info with correct name
+  ansible.builtin.assert:
     that:
-      - hcloud_images.hcloud_image_info | list | count == 1
-      - hcloud_images.hcloud_image_info[0].architecture == "arm"
+      - result.hcloud_image_info | list | count == 1
+      - result.hcloud_image_info[0].architecture == "arm"
 
-- name: test gather hcloud image infos with architecture
+- name: Gather hcloud_image_info with correct label selector
   hetzner.hcloud.hcloud_image_info:
-    architecture: arm
-  register: hcloud_images
-- name: verify test gather hcloud image infos with name
-  assert:
+    label_selector: "key=value"
+    type: snapshot
+  register: result
+- name: Verify hcloud_image_info with correct label selector
+  ansible.builtin.assert:
+    that:
+      # Snapshot names are stored in the description field
+      - >
+        result.hcloud_image_info
+        | selectattr('description', 'equalto', '{{ hcloud_snapshot_name }}')
+        | list | count == 1
+
+- name: Gather hcloud_image_info with wrong label selector
+  hetzner.hcloud.hcloud_image_info:
+    label_selector: "key!=value"
+    type: snapshot
+  register: result
+- name: Verify hcloud_image_info with wrong label selector
+  ansible.builtin.assert:
     that:
-      - hcloud_images.hcloud_image_info | selectattr('architecture','equalto','x86') | list | count == 0
-      - hcloud_images.hcloud_image_info | selectattr('architecture','equalto','arm') | list | count  > 2
+      - result.hcloud_image_info | list | count == 0
```

Comment:
This test might be flaky, `key!=value` applies to nearly all snapshots. Either because another test is currently creating a snapshot or when running this locally against a non-clean project.

We could check that our snapshot is not in the list of results, instead of checking `count == 0`

Might also be something for a follow up PR, as the test was included like this in the code before and multiple other tests have the same pattern.

Reply:
Yes, I'd rather postpone this change to another PR. We can then also make batch changes to all the `_info` targets.